src/live_restore/live_restore_fs.c

```diff
@@ -580,27 +589,41 @@ __live_restore_remove_extlist_hole(
     return (0);
 }
 
-/*
+typedef enum { FULL, PARTIAL, NONE } WT_LIVE_RESTORE_SERVICE_STATE;
+
+/* !!!
  * __live_restore_can_service_read --
- *     Return if a read can be serviced by the destination file. This assumes that the block manager
- *     is the only thing that perform reads and it only reads and writes full blocks. If that
- *     changes this code will unceremoniously fall over.
+ *     Return if a read can be serviced by the destination file. Callers must hold the extent list
+ *     read lock at a minimum.
+ *     There are three possible scenarios:
+ *     - The read is entirely within a hole and we return NONE.
+ *     - The read is entirely outside of all holes and we return FULL.
+ *     - The read begins outside a hole and then ends inside, in which case we return PARTIAL.
+ *       This scenario will only happen if background data migration occurs concurrently and has
+ *       partially the migrated content we're reading. The background threads always copies data in
```

Comment:
Still missing a word: "... and has partially the migrated content...". It is missing a verb: "copied", "moved", "requested", etc.

Reply:
How about "and has partially migrated the content we're reading."?

Reply:
f328ef403